src/main/java/com/faforever/server/map/Ladder1v1MapRepository.java

```diff
@@ -2,9 +2,13 @@
 
 import com.faforever.server.entity.MapVersion;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
 @Repository
 public interface Ladder1v1MapRepository extends JpaRepository<MapVersion, Integer> {
-
+  @Query("SELECT game_stats.mapId FROM game_player_stats JOIN game_stats WHERE playerId=?0 and game_stats.gameType=6 ORDER BY game_stats.startTime ASC LIMIT ?1")
+  List<MapVersion> findLastestLapperMapVersions(int playerId, int number);
```

Comment:
Yes, this is a native query. I doubt this works? You have to use JPQL.

And what is "Lapper"? :P
I'd call it `findRecentlyPlayedLadderMapVersions`

Reply:
Why do you doubt it?

Reply:
oh ok JQL